### PR TITLE
feat(query): show indexes add inverted indexes

### PR DESCRIPTION
**Pull request by b41sh**
I hereby agree to the terms of the CLA available at: https://docs.databend.com/dev/policies/cla/

## Summary

* show indexes add inverted indexes
* modify and drop column check if the column is referenced by inverted index

- part of #14825

## Tests

- [ ] Unit Test
- [x] Logic Test
- [ ] Benchmark Test
- [ ] No Test  - _Explain why_

## Type of change

- [ ] Bug Fix (non-breaking change which fixes an issue)
- [x] New Feature (non-breaking change which adds functionality)
- [ ] Breaking Change (fix or feature that could cause existing functionality not to work as expected)
- [ ] Documentation Update
- [ ] Refactoring
- [ ] Performance Improvement
- [ ] Other (please describe):

- - -
This change is [Reviewable](https://reviewable.io/reviews/datafuselabs/databend/15733)

### CODE REVIEW

File: src/query/service/src/test_kits/block_writer.rs [hunk 1]
```diff
@@ -90,6 +91,7 @@ impl<'a> BlockWriter<'a> {
             location,
             bloom_filter_index_location,
             bloom_filter_index_size,
+            inverted_index_size,
```
Comment (TCeason) [inline src/query/service/src/test_kits/block_writer.rs:94]: Maybe directly write a None? not use a new variable?

File: src/query/service/tests/it/storages/fuse/statistics.rs [hunk 2]
```diff
@@ -624,6 +625,7 @@ fn test_reduce_block_meta() -> databend_common_exception::Result<()> {
             location.clone(),
             None,
             bloom_filter_index_size,
+            inverted_index_size,
```
Comment (TCeason) [inline src/query/service/tests/it/storages/fuse/statistics.rs:628]: Ditto

File: src/query/storages/system/src/indexes_table.rs [hunk 3]
```diff
@@ -113,4 +148,66 @@ impl IndexesTable {
...
+                    ctx.push_warning(msg);
+                    continue;
+                }
+            };
+            for table in tables {
+                let table_info = table.get_table_info();
+                if table_info.meta.indexes.is_empty() {
+                    continue;
+                }
+                if visibility_checker.check_table_visibility(
```
Comment (TCeason) [inline src/query/storages/system/src/indexes_table.rs:200]: If use visibility checker better to add some test in `18_rbac`.

File: tests/suites/0_stateless/18_rbac/18_0011_index_visibility.sh [hunk 4]
```diff
@@ -0,0 +1,29 @@
...
+echo "create user u1 identified by '123' with DEFAULT_ROLE='role1';" | $BENDSQL_CLIENT_CONNECT
+echo "create user u2 identified by '123' with DEFAULT_ROLE='role2';" | $BENDSQL_CLIENT_CONNECT
+echo "grant role role1 to u1;" | $BENDSQL_CLIENT_CONNECT
+echo "grant role role2 to u2;" | $BENDSQL_CLIENT_CONNECT
+
+echo "=== test u1 with role1 ==="
+export TEST_U1_CONNECT="bendsql --user=u1 --password=123 --host=${QUERY_MYSQL_HANDLER_HOST} --port ${QUERY_HTTP_HANDLER_PORT}"
+echo "select name, type, definition from system.indexes order by name" | $TEST_U1_CONNECT
+
+echo "=== test u2 with role1 ==="
```
Comment (TCeason) [inline tests/suites/0_stateless/18_rbac/18_0011_index_visibility.sh:26]: "grant role role2 to u2;"

test u2 with role1 -> test u2 with role2

**Review by TCeason — APPROVED** [review]
Privilege part is LGTM

**Review by sundy-li — APPROVED** [review]

**Pull request merged**